### PR TITLE
feat: added uninstall of vsix extension for other visual studio versions

### LINKED ISSUE #73: Uninstall doesn’t remove vsix from VisualStudio(s), but only from the TcXaeShell

### PR DESCRIPTION
**Pull request by seehma**
this closes #73

### CODE REVIEW

File: Twinpack.iss [hunk 1]
```diff
@@ -268,4 +266,35 @@ begin
...
+  ReturnCode : Integer;
+begin
+  TwinpackVsixGuid := 'TwinpackVsix.26e0356d-ac0e-4e6a-a50d-dd2a812f6f23';
+
+  case CurUninstallStep of
+    usUninstall:
+      begin
+        for i:= 0 to DisplayNames.Count-1 do
+        begin
+          ShellExec('', InstallationPaths[i] + '\Common7\IDE\VSIXInstaller.exe', '/force ' + '/u:'+TwinpackVsixGuid+' /quiet', '', SW_HIDE, ewWaitUntilTerminated, ReturnCode);
```
Comment (iadonkey) [inline Twinpack.iss:292]: Do we need force?

File: Twinpack.iss [hunk 2]
```diff
@@ -38,7 +38,7 @@ InfoBeforeFile=DISCLAIMER
 [Files]
 Source: "TwinpackVsix\bin\{#MyConfiguration}\Package\*"; DestDir: "{#TcXaeShellExtensionsFolder}Zeugwerk\Twinpack"; Flags: ignoreversion recursesubdirs createallsubdirs; Check: InstallVsixInTcXaeShell;
 Source: "TwinpackVsix\bin\{#MyConfiguration}\TwinpackVsix.vsix"; DestDir: "{tmp}"; Flags: deleteafterinstall;
-Source: "vswhere.exe"; DestDir: "{tmp}"; Flags: deleteafterinstall;
+Source: "vswhere.exe"; DestDir: "{#TcXaeShellExtensionsFolder}Zeugwerk\Twinpack"; Flags: ignoreversion recursesubdirs createallsubdirs;
```
Comment (iadonkey) [inline Twinpack.iss:41]: Why store vswhere permanently, can’t you Extract to temp like it is done on install

Comment (iadonkey) [reply Twinpack.iss:41]: ExtractTemporaryFile('vswhere.exe');

File: Twinpack.iss [hunk 3]
```diff
@@ -253,12 +252,11 @@ begin
     begin
       if(VisualStudioOptionsPage.CheckListBox.Checked[i+1]) then
       begin
-        ShellExec('', InstallationPaths[i] + '\Common7\IDE\VSIXInstaller.exe', '/force ' + '/u:TwinpackVsix.26e0356d-ac0e-4e6a-a50d-dd2a812f6f23 /quiet', '', SW_HIDE, ewWaitUntilTerminated, ReturnCode);
+        ShellExec('', InstallationPaths[i] + '\Common7\IDE\VSIXInstaller.exe', '/force ' + '/u:'+TwinpackVsixGuid+' /quiet', '', SW_HIDE, ewWaitUntilTerminated, ReturnCode);
```
Comment (iadonkey) [inline Twinpack.iss:255]: Is force needed here?

File: Twinpack.iss (hunk 2 above)
Comment (seehma) [reply Twinpack.iss:41]: nope, does not work because extraction is only working with initial setup file

Comment (iadonkey) [reply Twinpack.iss:41]: Then maybe change it in the install part as well to use the local copy?

Comment (seehma) [reply Twinpack.iss:41]: at this time vswhere is not yet extracted to installation folder, i would leave it the way it is...

File: Twinpack.iss (hunk 1 above)
Comment (seehma) [reply Twinpack.iss]: nope -> removed

File: Twinpack.iss (hunk 3 above)
Comment (seehma) [reply Twinpack.iss]: nope -> removed

**Review by iadonkey — APPROVED** [review]

**Pull request merged**